coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/CoordinatorRuntimeTest.java

```diff
@@ -4292,6 +4294,185 @@ public void testRecordFlushTime() throws Exception {
         assertEquals("response3", write3.get(5, TimeUnit.SECONDS));
     }
 
+    @Test
+    public void testLargeCompressibleRecordTriggersFlushAndSucceeds() throws Exception {
+        MockTimer timer = new MockTimer();
+
+        // Writer sleeps for 10ms before appending records.
+        MockPartitionWriter writer = new MockPartitionWriter(timer.time(), Integer.MAX_VALUE, false);
+        CoordinatorRuntimeMetrics runtimeMetrics = mock(CoordinatorRuntimeMetrics.class);
```

Comment:
We don't need to mock timings or metrics in the new tests. We can use `testScheduleWriteOperationWithBatching` as a template for the new tests.
```suggestion
        MockTimer timer = new MockTimer();
        MockPartitionWriter writer = new MockPartitionWriter();
```